esx_service/vmdk_ops.py

```diff
@@ -1075,22 +1084,115 @@ def load_vmci():
    else:
        lib = CDLL(os.path.join(LIB_LOC, "libvmci_srv.so"), use_errno=True)
 
+def getLock(lockname):
+    '''
+    Create or return a existing lock identified by lockname.
+    '''
+    global rsrcLocks
+    global managerLock
+
+    if not rsrcLocks:
+        rsrcLocks = WeakValueDictionary()
+        logging.debug("getLock(): Initialized Lock store")
+    if not managerLock:
+        managerLock = threading.Lock()
+        logging.debug("getLock(): Initialized Manager Lock")
+    
+    with managerLock:
+        try:
+            lock = rsrcLocks[lockname]
+            logging.debug("getLock(): return existing lock %s", lockname)
+        except KeyError as e:
+            lock = threading.Lock()
+            rsrcLocks[lockname] = lock
+            logging.debug("getLock(): return new lock %s", lockname)
+    return lock
+
+def execRequestThread(clntsock, cartel, request):
+    '''
+    Execute requests in a thread context with a per volume locking. 
+    '''
+    # Before we start, block to allow main thread to advance.
+    time.sleep(0.001)
```

Comment:
I agree with Mark here. Thread scheduling is done by the OS. It's entirely possible that the main thread will not run before this thread resumes. I haven't looked at the rest of the code and am not sure why the main thread needs to run again first, but if it needs to be guaranteed. It'd be better to use some sort of signaling mechanism like a semaphore.